res/controllers/midi-components-0.0.js

```diff
@@ -789,6 +789,10 @@
         input: function(_channel, control, _value, status, _group) {
             throw "Called wrong input handler for " + status + ": " + control + ".\n" +
                 "Please bind jogwheel-related messages to inputWheel and inputTouch!\n";
+        },
+        connect: function() {
+            Component.prototype.connect.call(this);
+            this.deck = parseInt(script.channelRegEx.exec(this.group)[1]);
         }
```

Comment:
I think this needs an explanation.
```suggestion
        // this is needed for features such as "deck switching" that work
        // by changing the component group. It is assumed they call `connect`
        // afterwards.
        connect: function() {
            Component.prototype.connect.call(this);
            this.deck = parseInt(script.channelRegEx.exec(this.group)[1]);
        }
```